PetGame/Services/PetService.cs

```diff
@@ -298,31 +300,38 @@ private int CalculateHoursSinceLastActivity(IEnumerable<Activity> PastActivities
             return HoursSinceLastActivity;
         }
 
-        private int CalculateMinutesToNextAction(IEnumerable<Activity> PastActivities, int MinutesToNextAction, int HoursToCheck)
+        private DateTime TimeOfNextAction(IEnumerable<Activity> PastActivities, int MinutesToNextAction, int HoursToCheck)
         {
             List<Activity> Activities = (List<Activity>)PastActivities;
```

Comment:
```suggestion
             List<Activity> Activities = new List<Activity>(PastActivities);
```
Casting like this can be error prone.